src/redux/uniswapLiquidity.ts

```diff
@@ -0,0 +1,199 @@
+import { concat, filter, isEmpty, map, uniqBy } from 'lodash';
+import { AppState } from 'react-native';
+import { Dispatch } from 'redux';
+import { ThunkDispatch } from 'redux-thunk';
+import { emitChartsRequest } from './explorer';
+import { AppGetState } from './store';
+import { ParsedAddressAsset } from '@rainbow-me/entities';
+import {
+  getLiquidity,
+  saveLiquidity,
+} from '@rainbow-me/handlers/localstorage/uniswap';
+
+// -- Constants ------------------------------------------------------------- //
+
+const UNISWAP_UPDATE_LIQUIDITY_TOKENS =
+  'uniswap/UNISWAP_UPDATE_LIQUIDITY_TOKENS';
+const UNISWAP_LP_TOKENS_CLEAR_STATE = 'uniswap/UNISWAP_LP_TOKENS_CLEAR_STATE';
+
+const UNISWAP_POOLS_DETAILS = 'uniswap/UNISWAP_POOLS_DETAILS';
+
+// -- Actions --------------------------------------------------------------- //
+
+/**
+ * Details for a token within a Uniswap pool.
+ */
+interface UniswapPoolToken {
+  id: string;
+  name: string;
+  symbol: string;
+}
+
+/**
+ * Details for a loaded Uniswap pool.
+ */
+interface UniswapPoolAddressDetails {
+  address: string;
+  annualized_fees: number;
+  liquidity: number;
+  oneDayVolumeUSD: number;
+  profit30d: number;
```

Comment:
can be undefined: `profit30d?: number;`
according to: https://github.com/rainbow-me/rainbow/blob/develop/src/hooks/useUniswapPools.js#L99

Reply:
@jinchung Yes that's correct, and I actually also noticed that there are more fields that could be undefined due to `usePoolDetails` and `fetchPoolDetails`, which fetch only a subset of the fields: https://github.com/rainbow-me/rainbow/blob/develop/src/hooks/usePoolDetails.js#L20.

I ended up splitting it into two types, one for the data loaded by `useUniswapPools` and another for the subset loaded by `usePoolDetails`, since either one could be the source of the data as far as I can tell. Thanks!